### DISCUSSION TITLE
Renovate stuck with some branches (no-pending, already merged PR with same name)

### DISCUSSION THREAD

**Discussion by billy-mosse** [renovatebot/renovate#18632] [Request Help]
Hi all! I noticed that under certain circumstances I am wasting resources with renovate, and was wondering if there is an easy fix:

My configuration uses `"prCreation": "not-pending"`. In one of my repositories, there is an already merged PR associated with a branch called `renovate/docker`. When renovate runs, it generates a new candidate branch with that same name (with new upgrades).

It finds the merged branch and decides to perform a rebase:

```
DEBUG: Rebase branch cause the commits from old branch have already been merged. (repository=****, branch=renovate/docker)
DEBUG: Using reuseExistingBranch: false (repository=****, branch=renovate/docker)
```

It pushes a new commit to the branch, and marks its status as pending:
```
 INFO: Branch updated (repository=****, branch=renovate/docker)
       "commitSha": "****"
DEBUG: Branch status pending (repository=****, branch=renovate/docker)
       "commitSha": "****"
```

However, the build fails. When renovate runs again, it doesn't create the PR. Instead, the whole process gets repeated. In particular, the commit pushed in the previous renovate PR becomes an orphan - I see the following on GitHub if I click on the commit:

```
This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
```

To summarise, every time renovate runs (and it's set up to run frequently), the `renovate/docker` branch gets rewritten, with it having only 1 extra commit when compared to `main`.

I could fix the failing branch, but:
1. I'm interested in solving this problem at scale
2. Not having a PR created means that I have very little visibility on this problem (I need to check renovate logs or manually inspect branches)

I would appreciate any help. Happy to discuss and provide more information :). Thank you

**Comment by rarkins** [comment]
1. How are you running Renovate? e.g. self-hosted or the app?
2. Can you work out step by step instructions for how to reproduce this locally?

**Comment by billy-mosse** [comment]
I'm self-hosting renovate. In order to reproduce this, it's enough to:

1. 
```
{
  "prCreation": "not-pending"
}
```

2. Run renovate on a repository with an outdated dependency.
3. Merge one of the PR it generates.
4. Restore the branch if renovate deleted it.
5. Downgrade the same dependency.
6. Force a test to fail.
7. Run renovate again, twice. The branch will get processed each time but no PR will be created (but the expected behaviour is, AFAIK, that a PR is indeed created).

Is this enough or should I also provide with a public playground github repository?

**Reply by rarkins** [reply]
GitHub would be ideal
